bin/run-lint

```diff
@@ -1,21 +1,42 @@
 #!/usr/bin/env node
-const child_process = require( 'child_process' );
+
 const path = require( 'path' );
-var args = [ '--cache', '--quiet', '--ext=.js',  '--ext=.jsx' ];
+const child_process = require( 'child_process' );
+const stylish = require( 'eslint/lib/formatters/stylish' );
+const exitCodeFromEslint = require( './eslint-formatters-calypso/src/lib/exit-code' );
+
+var args = [ '--cache', '--quiet', '--ext=.js', '--ext=.jsx' ];
+
+const argsBranchName = [ 'rev-parse', '--abbrev-ref', 'HEAD' ];
+const branchName = child_process.spawnSync( 'git', argsBranchName ).stdout.toString().trim();
+
+if ( branchName === 'master' ) {
```

Comment:
TL;DR:

* It can't be a plugin, but we need something more than a formatter (at least, the ability to control `exit codes`). This is something we can easily do by piping `eslint . | eslines`.
* Take advantage of this work from the editors seems easy, as they usually run `npm run lint/eslint` for linting, if the script exists in the package.json or allow to configure a command manually - I need to research more on this, though.

#### # More context about formatters VS plugins

An ESLint [plugin](http://eslint.org/docs/developer-guide/working-with-plugins) bundles a set of rules, environment variables and processors. An ESLint formatter is something you pass as an argument to the CLI and there is no way to make it play together with the config file (see [code](https://github.com/eslint/eslint/blob/master/lib/cli.js#L192) - `currentOptions.format` is the command line option `--format`). The ESLint team has already had some discussions around this: [move formatters into plugins](https://github.com/eslint/eslint/issues/3013#issuecomment-121780989) or [have the ability to pass options to formatters](https://github.com/eslint/eslint/issues/2989).

#### # More than a formatter

There are situations where people needs to configure the formatter (for example, to configure the colors or the file paths for a specific environment) or it needs to do more than an ESLint formatter is capable of.

To work around this, sometimes, people encapsulates the call to ESLint in its own script ([eslint-nibble](https://github.com/IanVS/eslint-nibble), [eslint-pretty](https://github.com/bugeats/eslint-pretty)).

Other approaches that I have seen in practice while using ESLint directly are:

The **recommended way** [is](https://github.com/eslint/eslint/issues/3013#issuecomment-122032581):

    eslint -f json . | formatter --your-options

This is something we can benefit from as it would allow us to control the `exit code` (actually, we can already do this, as before it was migrated to formatter this is the approach I was taking).

The **common way** to pass info to the formatter is by using environment variables or a config file (we use also the config file).

There is some other **hacky way** that depends on the fact that ESLint uses [optionator](https://www.npmjs.com/package/optionator) for parsing options from the command line. This is used by [eslint-friendly-formatter](https://github.com/royriojas/eslint-friendly-formatter#formatter-parameters). The way this works is:

* ESLint uses optionator to parse command line arguments.
* Optionator provides an API to do `optionator.parse(process.argv)` (see [ESLint code](https://github.com/eslint/eslint/blob/master/lib/cli.js#L135)) that returns an object like `{opt1: 2, opt2:3, _: ['positional-param-1', 'positional-param-2']}` with keys being the options and any other [positional parameter](http://wiki.bash-hackers.org/scripting/posparams) added in the `_` array.
* ESLint uses the contents in the `_` array as the input files (but [discards any file that does not exist](https://github.com/eslint/eslint/blob/master/lib/util/glob-util.js#L165)).
* Optionator happen to follow the bash convention of using `--` as a marker for _[end of options](http://wiki.bash-hackers.org/dict/terms/end_of_options)_ so any value that follows the marker is considered a [positional parameter](http://wiki.bash-hackers.org/scripting/posparams). _example of the hack:_

    command --opt1 2 --opt2 3 -- file-to-process-1 file-to-process-2 --opt3

    for this command, `optionator.parse(process.argv)` would return`{opt1: 2, opt2:3, _: ['file-to-process-1', 'file-to-process-2', '--opt3']}`.

* eslint-friendly-formatter takes advantage of this hack (ESLint would not process --opt3 as that file does not exists) to [retrieve the command line options after the `--`](https://github.com/royriojas/eslint-friendly-formatter/blob/master/index.js#L99) marker and use them for its own configuration.